app/wasm/bindings/query.go

```diff
@@ -0,0 +1,73 @@
+package wasmbindings
+
+import (
+	wasmvmtypes "github.com/CosmWasm/wasmvm/types"
+)
+
+// OsmosisQuery contains osmosis custom queries.
+// See https://github.com/confio/osmosis-bindings/blob/main/packages/bindings/src/query.rs
+type OsmosisQuery struct {
+	// /// Given a subdenom minted by a contract via `OsmosisMsg::MintTokens`,
+	// /// returns the full denom as used by `BankMsg::Send`.
+	// FullDenom *FullDenom `json:"full_denom,omitempty"`
+	/// For a given pool ID, list all tokens traded on it with current liquidity (spot).
+	/// As well as the total number of LP shares and their denom.
+	PoolState *PoolState `json:"pool_state,omitempty"`
```

Comment:
Should we rename these types to have a Query suffix?

Reply:
Is that the norm across the project? If not, I would leave it as it is.

Reply:
That is the norm for all of the golang queries, I don't know whats the norm for the cosmwasm side

Reply:
We typically label them the same name as the field.

Again, as long as the json stays the same, you can gladly rename the types as you wish in another PR.
These are for use inside the osmosis Golang code, so it should be whatever idiom looks good to you.